api/src/main/java/jakarta/config/Config.java

```diff
@@ -30,27 +28,73 @@
  */
 public interface Config {
     /**
-     * Return all the currently registered {@link jakarta.config.spi.ConfigSource sources} for this configuration.
+     * Context related to the root configuration instance.
+     *
+     * @return configuration context
+     */
+    ConfigContext context();
+
+    /**
+     * Fully qualified key of this config node (such as {@code server.port}).
+     * Returns an empty String for root config.
+     *
+     * @return key of this config
+     */
+    String getKey();
+
+    /**
+     * Name of this node - the last element of a fully qualified key.
      * <p>
-     * The returned sources will be sorted by priority and name, which can be iterated in a thread-safe
-     * manner. The {@link java.lang.Iterable Iterable} contains a fixed number of {@link jakarta.config.spi.ConfigSource
-     * configuration
-     * sources}, determined at application start time, and the config sources themselves may be static or dynamic.
+     * For example for key {@code server.port} this method would return {@code port}.
+     *
+     * @return name of this node
+     */
+    String getName();
```

Comment:
I think it is confusing to have both `getKey()` and `getName()`. What use case requires to retrive the last part of the key? Besides the method names are very confusing.

Reply:
renamed to `getNodeName()`